ocrd_tesserocr/common.py

```diff
@@ -128,17 +261,28 @@ def image_from_region(workspace, region,
         region_image = workspace.resolve_image_as_pil(
             alternative_image[-1].get_filename())
     else:
-        region_image = page_image.crop(
+        # create a mask from the region polygon:
+        region_image = image_from_polygon(page_image, region_polygon)
+        # recrop into region rectangle:
+        region_image = crop_image(region_image,
             box=(region_xywh['x'] - page_xywh['x'],
                  region_xywh['y'] - page_xywh['y'],
                  region_xywh['x'] - page_xywh['x'] + region_xywh['w'],
                  region_xywh['y'] - page_xywh['y'] + region_xywh['h']))
-        # FIXME: mask any overlapping regions (esp. Separator/Noise/Image)
-        # but we might need overlapping rules: e.g. an ImageRegion which
+        # note: We should mask overlapping neighbouring regions here
+        # (especially Separator/Noise/GraphicRegion, but also TextRegion),
+        # but finding the right clipping rules can be difficult if operating
+        # on the raw (non-binary) image data alone: e.g. an ImageRegion which
         # properly contains our TextRegion should be completely ignored, but
         # an ImageRegion which is properly contained in our TextRegion should
         # be completely masked, while partial overlap may be more difficult
-        # to decide (use polygons?)
+        # to decide. On the other hand, on the binary image, we can use
+        # connected component analysis to mask foreground areas which originate
+        # in the neighbouring regions. But that would introduce either the
+        # assumption that the input has already been binarized, or a dependency
+        # on some ad-hoc binarization method. Thus, it is preferable to use
+        # a dedicated processor for this (which produces clipped AlternativeImage
+        # or reduced polygon coordinates).
```

Comment:
To be discussed live.

Reply:
Resolved alright? Feel free to post our result here (I don't quite remember).